package/MDAnalysis/coordinates/chain.py

```diff
@@ -397,4 +397,27 @@ def __repr__(self):
                     nframes=self.n_frames,
                     natoms=self.n_atoms))
 
-
+    def add_transformations(self, transform_list):
+        """ Adds all the transformations to be applied to the trajectory.
+        Overrides :meth:`~MDAnalysis.coordinates.base.ProtoReader.add_transformations`
+        to avoid unintended behaviour where the coordinates of each frame are transformed
+        multiple times when iterating over the trajectory.
+        
+        In this method the transformations are added directly to the reader instances and,
+        thus, will be applied correctly.
+        """
+        try:
```

Comment:
As in the `SingleFrameReaderBase` case above `super` could be used here and reduce the duplication of these checks (calling the `add_transformations` method from `ProtoReader` is safe here, and in `SingleFrameReaderBase`, because the `_apply_transformations` method of these classes is just a passthrough, anyway).